### PR TITLE
fix: Android scan improvement

### PR DESCRIPTION
**Pull request by Dmitry-Borodin**
## Purpose
Improve scanning performance on some devices where it's very poor

## Scope
- Required resolution is changed - default is 640*480 and it's not enough
- analyzing in background thread as big images taking a lot of CPU to analyze
- paddings for frame increased so users won't keep phone too close to the screen. 40.dp as it was before

This targeting to improve #1762
Feedback on scanning on different devices is very welcome
Build with those changes to download - https://drive.google.com/file/d/17YWhZIZgY8JM6HzsYAFxe2tUzKlCGWFj/view?usp=share_link

### CODE REVIEW

File: android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt [hunk 1]
```diff
@@ -27,6 +32,10 @@ internal fun CameraViewInternal(viewModel: CameraViewModel) {
 	val cameraProviderFuture =
 		remember { ProcessCameraProvider.getInstance(context) }
 	val coroutineScope = rememberCoroutineScope()
+	val configuration = LocalConfiguration.current
+val backgroundExecutor = remember {
```
Comment (krodak) [inline android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt:36]:
```suggestion
    val backgroundExecutor = remember {
```
But seems like something that linter / formatter should catch

File: android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt [hunk 2]
```diff
@@ -57,19 +68,32 @@ internal fun CameraViewInternal(viewModel: CameraViewModel) {
...
+					.setOutputImageFormat(ImageAnalysis.OUTPUT_IMAGE_FORMAT_YUV_420_888)
+					.setTargetResolution(Size(1920,1080))
 					.build()
 					.apply {
 						setAnalyzer(executor) { imageProxy ->
 							viewModel.processFrame(barcodeScanner, imageProxy)
 						}
 					}
 
+				Log.e("rational is ", "${configuration.screenWidthDp} ${configuration.screenHeightDp}")
```
Comment (krodak) [inline android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt:80]: We probably shouldn't leave that log here

**Review by krodak — APPROVED** [review]

File: android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt (hunk 2 above)
Comment (Dmitry-Borodin) [reply android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt:80]: fixed

File: android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt (hunk 1 above)
Comment (Dmitry-Borodin) [reply android/src/main/java/io/parity/signer/screens/scan/camera/CameraView.kt:36]: we don't have automated formatter, formatted.

File: android/src/main/java/io/parity/signer/screens/scan/camera/CameraViewModel.kt [hunk 3]
```diff
@@ -44,6 +46,7 @@ class CameraViewModel() : ViewModel() {
 		barcodeScanner: BarcodeScanner,
 		imageProxy: ImageProxy
 	) {
+
```
[the remark this answers is not in the document]
Comment (Dmitry-Borodin) [reply android/src/main/java/io/parity/signer/screens/scan/camera/CameraViewModel.kt:49]: Just semantic separation of different sections. I think this is ok.
This part was changed few times during changes and will probably be changed in future optimisations

Thanks for review.

**Comment by Tbaut** [comment]
Thanks, this fixes the issue mentioned in #1762 
I've had the best results with the phone actually pretty far from the screen, something like this

sign

**Pull request merged**